cartography/intel/aws/cloudfront.py

```diff
@@ -0,0 +1,201 @@
+
+logger = logging.getLogger(__name__)
+stat_handler = get_stats_client(__name__)
+
+
+@timeit
+@aws_handle_regions
+def get_cloudfront_distributions(boto3_session: boto3.session.Session) -> List[Dict]:
+    client = boto3_session.client('cloudfront')
+    paginator = client.get_paginator('list_distributions')
+    cloudfront_distributions = []
+    for page in paginator.paginate():
+        for distribution in page['DistributionList']['Items']:
+            cloudfront_distributions.append(distribution)
+    return cloudfront_distributions
+
+
+@timeit
+def load_distribution_origin(
+        neo4j_session: neo4j.Session,
+        distribution_arn: str,
+        origins: Dict,
+        current_aws_account_id: str,
+        aws_update_tag: int,
+) -> None:
+    ingest_definitions = """
+        MERGE (d:CloudFrontDistributionOrigin{id: $distribution_arn + "-" + $origin.Id})
+        ON CREATE SET d.firstseen = timestamp()
+        SET
+            d.domain_name = $origin.DomainName,
+            d.origin_path = $origin.OriginPath,
+            d.custom_headers = $origin.CustomHeaders.Items,
+            d.connection_timeout = $origin.ConnectionTimeout,
+            d.connection_attempts = $origin.ConnectionAttempts,
+            d.origin_access_control_id = $origin.OriginAccessControlId,
+            d.originshield_enabled = $origin.OriginShield.Enabled,
+            d.originshield_region = $origin.OriginShield.OriginShieldRegion,
+            d.s3_origin_enabled = $origin.S3OriginConfig.S3OriginConfigEnabled,
+            d.custom_origin_enabled = $origin.CustomOriginConfig.CustomOriginConfigEnabled,
+            d.s3_origin_config = $origin.S3OriginConfig_Data,
+            d.custom_origin_config = $origin.CustomOriginConfig_Data,
+            d.lastupdated = $aws_update_tag
+        WITH d
+        MATCH (td:CloudFrontDistribution{arn: $distribution_arn})
+        MERGE (td)-[r:HAS_ORIGIN]->(d)
+        ON CREATE SET r.firstseen = timestamp()
+        SET r.lastupdated = $aws_update_tag
+        """
+
+    for origin in origins:
+        if origin["CustomHeaders"]["Quantity"] == 0:
+            origin["CustomHeaders"]["Items"] = []
+
+        origin["CustomHeaders"]["Items"] = json.dumps(origin["CustomHeaders"]["Items"])
+
+        if not origin["OriginShield"]["Enabled"]:
+            origin["OriginShield"]["OriginShieldRegion"] = ""
+
+        if "S3OriginConfig" not in origin:
+            origin["S3OriginConfig"] = {}
+            origin["S3OriginConfig"]["S3OriginConfigEnabled"] = False
+        else:
+            origin["S3OriginConfig"]["S3OriginConfigEnabled"] = True
+
+        if "CustomOriginConfig" not in origin:
+            origin["CustomOriginConfig"] = {}
+            origin["CustomOriginConfig"]["CustomOriginConfigEnabled"] = False
+        else:
+            origin["CustomOriginConfig"]["CustomOriginConfigEnabled"] = True
+
+        origin["S3OriginConfig_Data"] = json.dumps(origin["S3OriginConfig"])
```

Comment:
Let's not dump jsons into node properties. Are these objects too complex to add their properties directly to the node? 

- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-s3originconfig.html
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cloudfront-distribution-customoriginconfig.html